lib/IRGen/IRGenSIL.cpp

```diff
@@ -5509,6 +5510,43 @@ void IRGenSILFunction::visitSuperMethodInst(swift::SuperMethodInst *i) {
   auto method = i->getMember();
   auto methodType = i->getType().castTo<SILFunctionType>();
 
+  auto baseMethod = method.getOverriddenVTableEntry();
+  auto *classDecl = cast<ClassDecl>(baseMethod.getDecl()->getDeclContext());
+  if (IGM.isResilient(classDecl, ResilienceExpansion::Maximal)) {
+    // Load the superclass of the instance class resiliently.
+    //
+    // FIXME: Perhaps if we knew that the instance type of the base value
+    // was exact, we could avoid this.
```

Comment:
Specifically, we can avoid this part when in the same resilience domain as the *current* class. But that's an optimization, so it can come later.

Reply:
Yeah, the instanceTy here is the “current” class. I’ll clarify the comments

Reply:
Ok, I get what you mean. The static type of 'self' might be in our resilience domain in which case we can assume what its superclass is, even if the ancestor that _defines_ the method is in a different resilience domain. This is simple enough to implement; I'll do it in this PR.